### PR TITLE
PR to view/comment on Prihan/Marlin changes

### PR DESCRIPTION
**Pull request by ex-nerd**
I've been cherry-picking changes from @Prihan's `deltabot-dev` branch but figured a pull request would be useful for sharing/commenting as I go.

### CODE REVIEW

**Comment by ex-nerd** [comment]
@Prihan I've cherry-picked a few of your changes but in the process broke the merge (changed a couple of your comments, etc).  Just thought I'd ping you here so you can start keeping an eye on this PR.

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -400,10 +418,10 @@ const bool Z_MAX_ENDSTOP_INVERTING = false; // set to true to invert the logic o
 #define BED_CENTER_AT_0_0  // If defined, the center of the bed is at (X=0, Y=0)
 
 //Manual homing switch locations:
-// For deltabots this means top and center of the Cartesian print volume.
+// For deltabots this means top and center of the cartesian print volume.
 #define MANUAL_X_HOME_POS 0
 #define MANUAL_Y_HOME_POS 0
-#define MANUAL_Z_HOME_POS 300  // For delta: Distance between nozzle and print surface after homing.
+#define MANUAL_Z_HOME_POS 280.67 //300  // For delta: Distance between nozzle and print surface after homing.
```
Comment (ex-nerd) [inline Marlin/Configuration.h:424]: Any particular position you changed this, other than to better approximate the kossel pro itself?  I'm just curious if it provides any actual benefit other than possibly preventing a head crash for someone who doesn't always run G29.

Comment (Prihan) [inline Marlin/Configuration.h:424]: Two reasons - it's a better fit for the KosselPro as well as a closer approximation for RichCattell's autocal routines to start from.  The process will set the height such that any endstop adjustment needed will be negative and will tweak further during the process such that the bed height tends to 0.00.

After the process you can fine-tune M666 Hn.nn to set the head clearance correctly.

**Comment by Prihan** [comment]
The trouble with this change is that it is pretty monolithic :)

I guess it splits into four major areas:

Changes to Configuration/Configuration_Adv.h to reflect the additional adjustments for auto-calibration and tune some of the parameters (these are not required for auto-calibration but made me feel better ...)

Changes to the EEPROM stored variables to support the autocal. Most of this is in Configuration_store.cpp but some global variables and procedure definitions are added to Marlin.h and defined in Marlin_Main.cpp

The bulk of the changes are in Marlin_Main.cpp. A huge chunk of this is the autocal G30 and M666 routines lifted from RichCattell's Marlin/Testing branch.  I have adapted them slightly to work with the existing zprobe code and added supporting variables and functions. There is probably still some redundancy here and we may be able to simplify a bit.

Finally the M48 changes are relatively standalone within Marlin_Main.cpp and add delta-safe movement and probing to the existing code. My second commit added some additional stats to the result.

**Comment by ex-nerd** [comment]
monolithic..  yes.  That's why I started cherry-picking specific changes.  I was hoping that when I did so they would disappear from the diff but apparently github just gives up when there's a conflict and shows the diff against the version of this repo that the PR was created against, rather than the latest version.

**Comment by ex-nerd** [comment]
Any more thoughts on this?  I'd love to be able to progress further than we are here before trying to merge these changes with the upstream config changes.

**Comment by Prihan** [comment]
I am not sure what you are looking for here.

Because this is a significant new function you have to make most of the changes in one hit. If I had to break it down into functional chunks I would break it down as follows:
- The configuration_store (EEPROM) changes first as a standalone supporting change but this involves some supporting variables and functions in Marlin_main and some configuration.h defaults changes.
- Follow up with the M665 etc functions to support the EEPROM changes which will use the read/write routines above
- Add the G30 supporting functions and tweaks to probing routines which should not affect existing functionality.
- Add the G30 function itself
- Add the M48 probe repeatability changes 

It is complicated by the fact that some of the parameters that may have been fixed when read from EEPROM or defaults are now changeable on the fly and this has knock-on effects throughout the code.

To be honest the way I approached this was to configure Rich's code for Kosselpro and get that working and then mega-merge the relevant files into your openbeam build and resolve conflicts followed up by a cleanup and bug squashing pass (or six!) until the basic G30 and EEPROM stuff was working. The M48 stuff came later after I crashed my head a couple of times trying it out. As a result I don't have a set of nice clean incremental changes to offer :)

Peter

**Comment by ex-nerd** [comment]
There were a bunch of small changes that I cherry-picked one by one, but between that and other changes that got pulled down, this branch no longer merges cleanly and I can't even see what has/hasn't been applied.

Then again, maybe it's all moot if there is going to be a separate effort getting these applied against upstream (which is a really messy merge into this branch and way over my head).

**Comment by Prihan** [comment]
Maybe if I re-base on your current version that would clean things up a bit? Either way I suspect another pull request is the easiest way - my Git-fu is not strong :(

**Comment by ex-nerd** [comment]
If you rebase and force-push your branch back up, this PR will auto-update.  A merge would do the same.  You could also rebase and push up to a new branch and create a new PR.

**Comment by Prihan** [comment]
I will take a look when I get a chance but I am away at the weekend for a few days so it might not be until later next week.

**Comment by wingrunr21** [comment]
I fixed the `deltabot-dev` branch to merge cleanly with the current `deltabot` in #6

**Comment by ex-nerd** [comment]
I'm not in a place where I can actually check this (started the lengthy process of getting things put onto Marlin's main branch, which required upgrading to Arduino 1.6, etc...).  Maybe @unrepentantgeek will have some time to check on this, though.

**Comment by ex-nerd** [comment]
Closing this since I haven't looked at this code in over 8 years.

**Pull request closed without merging**